WalletWasabi/Legal/LegalDocuments.cs

```diff
@@ -10,63 +10,84 @@ namespace WalletWasabi.Legal
 {
 	public class LegalDocuments
 	{
-		public const string EmbeddedFileName = "LegalDocuments.txt";
-		public const string LegalFolderName = "Legal";
-		public const string AssetsFoldername = "Assets";
-		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), LegalFolderName, AssetsFoldername, EmbeddedFileName);
+		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), "Legal", "Assets", "LegalDocuments.txt");
 
-		public LegalDocuments(string filePath)
+		public LegalDocuments(Version version, string content)
 		{
-			FilePath = Guard.NotNullOrEmptyOrWhitespace(nameof(filePath), filePath, trim: true);
-			var verString = Path.GetFileNameWithoutExtension(FilePath);
-			Version = Version.Parse(verString);
+			Version = version;
+			Content = content;
 		}
 
-		public string FilePath { get; }
 		public Version Version { get; }
+		public string Content { get; }
 
-		public static LegalDocuments TryLoadAgreed(string dataDir)
+		public static async Task<LegalDocuments?> TryLoadAgreedAsync(string folderPath)
 		{
-			var legalFolderPath = Path.Combine(dataDir, LegalFolderName);
-			IoHelpers.EnsureDirectoryExists(legalFolderPath);
-			var legalDocCandidates = FindCandidates(legalFolderPath);
+			if (!Directory.Exists(folderPath))
+			{
+				return null;
+			}
+
+			var legalDocCandidates = FindCandidates(folderPath);
```

Comment:
I would consider using `switch`. Then you don't need a temporary variable.

```csharp
switch (legalDocCandidates.Count())
{
  case 1:
    {
      var filePath = legalDocCandidates.Single();

      var content = await File.ReadAllTextAsync(filePath).ConfigureAwait(false);
      var verString = Path.GetFileNameWithoutExtension(filePath);
      var version = Version.Parse(verString);

      return new LegalDocuments(version, content);
    }

  case > 1:
    RemoveCandidates(folderPath);
    break;
}
```

Reply:
I luv it.